src/latexify/ast_utils.py

```diff
@@ -147,3 +147,9 @@ def extract_function_name_or_none(node: ast.Call) -> str | None:
         return node.func.attr
 
     return None
+
+
+def ast_function_def(*args, **kwargs) -> ast.FunctionDef:
+    if sys.version_info.minor < 12:
+        kwargs.pop("type_params", None)
+    return ast.FunctionDef(*args, **kwargs)
```

Comment:
I think we haven't use `type_params` anywhere with meaningful semantics, and this function can be defined to hide this variable from outer functions rather than passing it always.

Reply:
We need `type_params` to pass the `mypy` scan

Reply:
How about:

```python
def this_fn(...):
    if sys.version_info.minor < 12:
        # We know that the following invocation is correct.
        # In this case the mypy error is just wrong and we need to ignore it.
        return ast.FunctionDef(...)  # type: ignore[some-appropriate-flag]
  
    return ast.FunctionDef(..., type_params=[])
```

Reply:
I think mark ignore on this function is better. Let me update

Reply:
I updated follow your comment